src/components/graphqlFormGenerator/FormInput.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <!-- NOTE: the is field comes from `props` -->
+  <!-- eslint-disable-next-line vue/require-component-is -->
+  <component
+   v-model="model"
+   v-bind="props"
+   :label="label"
+   :gqlType="gqlType"
+  />
+</template>
+
+<script>
+import Vue from 'vue'
+
+import { VTextField } from 'vuetify/lib/components/VTextField'
+
+import { formElement } from '@/components/graphqlFormGenerator/mixins'
+import GNonNull from '@/components/graphqlFormGenerator/components/NonNull'
+import GList from '@/components/graphqlFormGenerator/components/List'
+
+/* Vuetify number input component.
+ *
+ * Note: Vuetify doesn't supply a dedicated number field, instead you
+ *       specialise the text field using `type='number'`, this, however,
+ *       does not cast values to `Number` for you so this extension parses
+ *       values to `Number` so they can be used directly in the data model.
+ */
+const VNumberField = Vue.component(
+  'v-number-field',
+  {
+    extends: VTextField,
+    computed: {
+      internalValue: {
+        get () {
+          return this.lazyValue
+        },
+        set (val) {
+          // cast values on `set` operations, note this does not get
+          // called on creation
+          this.lazyValue = Number(val)
+          this.$emit('input', this.lazyValue)
+        }
+      }
+    }
+  })
+
+// default props for all form inputs
+const DEFAULT_PROPS = {
+  filled: true,
+  rounded: true
+}
+
+// registry of GraphQL "named types" (e.g. String)
+// {namedType: {is: ComponentClass, prop1: value, ...}}
+const NAMED_TYPES = {
+  String: {
+    is: VTextField
+  },
+  Int: {
+    is: VNumberField,
+    type: 'number',
+    rules: [
+      // TODO: this should work but doesn't seem to be doing anything
+      x => Number.isInteger(x) || 'Integer'
+    ]
+  }
+}
+
+// registry of GraphQL "kinds" (e.g. LIST)
+// {namedType: {is: ComponentClass, prop1: value, ...}}
+const KINDS = {
+  NON_NULL: {
+    is: GNonNull
+  },
+  LIST: {
+    is: GList
+  }
+}
+
+export default {
+  name: 'g-form-input',
+
+  mixins: [formElement],
+
+  props: {
+    // dictionary of props for overriding default values
+    propOverrides: {
+      type: Object,
+      default: () => {}
+    }
+  },
+
+  data: () => ({
+    defaultProps: DEFAULT_PROPS
+  }),
+
+  computed: {
+    /* The props to pass to the form input.
```

Comment:
props = properties? Is this an ES5/6 standard jargon, or a project jargon? If it's project can you put the full word in the docstring?

Reply:
"Props" is special Vue.js jargon. Props are like arguments that you pass to a component e.g:

```html
<component prop1="foo" prop2="bar" />
```

They aren't quite like "properties" in Python, the component shouldn't change them. Vue "computed" values are more like Python `@properties`.

In a way props kinda belong to their parent component:

```vue
<template>
    <component :foo="foo" :bar="bar" />
</template>

<script>
export default {
  data: () => ({
    foo: 'Foot',
    bar: 'Bart'
  }).
  methods: {
    whatever () {
      // this component is free to make changes to foo and bar because these values are set
      // in `data` whereas in `<component />` they will be set as `props`.
      foo = 'Fooooot'
    }
  }
}
</script>
```